estimate_allowed_deviance.py

```diff
@@ -61,13 +66,31 @@ def _read_resutls(self):
         Try to load all json files with performance tests from directory
         :return: None
         """
-        path = self.options['directory']
+        path = self.options.directory
         if path is None:
-            path = getcwd()
-        file_list = [path + f for f in listdir(path) if isfile(join(path, f))]
+            path = os.getcwd() + os.path.sep
+        file_list = [path + f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]
         for filename in file_list:
             self._add_perf_test_result(filename)
 
+    def _estimate(self):
+        """
+        Compute estimation
+        :return: None
+        """
+        # if else statement is for newbies ;-)
+        {
+            'deviances': self._estimate_deviances,
+            'baseline': self._estimate_baseline
+        }[self.options.type]()
```

Comment:
For this case it is little bit overkill, but when you want avoid several if, elif, elif, elif, ... then it is IMHO right approach.